commands/start-container.ts

```diff
@@ -22,13 +24,33 @@ export function startContainerInteractive() {
 }
 
 export function startAzureCLI() {
-    
-    let option: string = process.platform === 'linux' ? '--net=host' : '';
-    let cmd: string = `docker run ${option} -it --rm azuresdk/azure-cli-python:latest`;
-
-    let terminal: vscode.Terminal = vscode.window.createTerminal('Azure CLI');
-    terminal.sendText(cmd);
-    terminal.show();
-    terminal.sendText(`az login`);
+    // block of we are on windows running windows containers... 
+    if (process.platform === 'win32') {
+        docker.getEngineType().then((engineType: DockerEngineType) => {
+            if (engineType === DockerEngineType.Windows) {
+                vscode.window.showErrorMessage<vscode.MessageItem>('Currently, you can only run the Azure CLI when running Linux based containers.',
+                    {
+                        title: 'More Information',
+                    },
+                    {
+                        title: 'Close',
+                        isCloseAffordance: true
+                    }
+                ).then((selected) => {
+                    if (!selected || selected.isCloseAffordance) {
+                        return;
+                    }
+                    return cp.exec('start https://docs.docker.com/docker-for-windows/#/switch-between-windows-and-linux-containers');
+                })
+            };
+        });
+    } else {
+        let option: string = process.platform === 'linux' ? '--net=host' : '';
+        let cmd: string = `docker run ${option} -it --rm azuresdk/azure-cli-python:latest`;
```

Comment:
I'm not sure whether it's a good idea, but we could potentially mound the `~/.azure` directory into the container, so that your login sessions and configuration settings (e.g. default output type, cloud type) persisted across runs of this command.